packages/contracts-bedrock/contracts/L2/L1Block.sol

```diff
@@ -74,5 +83,30 @@ contract L1Block is Semver {
         batcherHash = _batcherHash;
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
+
+        hashes.set(_number, _hash);
+    }
+
+    /// @notice Returns the hash of the L1 block at the specified block number.
+    /// @dev Requires the block number to be within the range of stored block hashes.
+    /// @param _number The number of the block for which to return the hash.
+    /// @return hash_ The hash of the L1 block.
+    function getL1BlockHash(uint64 _number) external view returns (bytes32 hash_) {
+        uint64 lastNumber = number;
+
+        if (_number > lastNumber) {
```

Comment:
Can we add some comments above each of these statement? Like: handle the case where the block is in the future or handle the case where the block is no longer in the ring buffer